### ISSUE TITLE
HBO Max

### ISSUE THREAD

**Issue by 89z** [ytdl-org/youtube-dl#31100]
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

https://www.hbomax.com/collections/watch-free/barry-s1-e1

## Description

Some videos are free, and it seems no DRM as well.

~~~
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.hbomax.com/collections/watch-free/barry-s1-e1']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.10.4 (CPython) - Windows-10-10.0.18363-SP0
[debug] exe versions: ffmpeg 2022-06-09-git-5d5a014199-essentials_build-www.gyan.dev
[debug] Proxy map: {}
[generic] barry-s1-e1: Requesting header
WARNING: Falling back on generic information extractor.
[generic] barry-s1-e1: Downloading webpage
[generic] barry-s1-e1: Extracting information
ERROR: Unsupported URL: https://www.hbomax.com/collections/watch-free/barry-s1-e1
Traceback (most recent call last):
  File "C:\python\lib\site-packages\youtube_dl\YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "C:\python\lib\site-packages\youtube_dl\YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "C:\python\lib\site-packages\youtube_dl\extractor\common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "C:\python\lib\site-packages\youtube_dl\extractor\generic.py", line 3489, in _real_extract
    raise UnsupportedError(url)
youtube_dl.utils.UnsupportedError: Unsupported URL: https://www.hbomax.com/collections/watch-free/barry-s1-e1
~~~

**Comment by gamer191** [comment]
> Some videos are free, and it seems no DRM as well.

What do you mean by "it seems"? Did you test whether it had drm? You can find instructions to do that [here](https://github.com/yt-dlp/yt-dlp/issues/4062#issuecomment-1153474056)

**Comment by 89z** [comment]
I might have more experience with DRM than anyone on this issue tracker, including @gamer191, so I take the previous comment as an insult.

I have now blocked @gamer191, because of their past history of trolling threads. I refuse to engage in any pointless back and forth with this person. They have a repeated history of derailing threads with off topic and political comments. Not gonna happen this time.

**Comment by SavSanta** [comment]
I must say thanks for the education @gamer191. That's good information

**Comment by 89z** [comment]
@SavSanta has been blocked. anyone else who wants to be blocked, feel free to mention @gamer191

**Comment by tacohitbox** [comment]
> > Some videos are free, and it seems no DRM as well.
> 
> What do you mean by "it seems"? Did you test whether it had drm? You can find instructions to do that [here](https://github.com/yt-dlp/yt-dlp/issues/4062#issuecomment-1153474056)

I got the m3u8 from one of the requests on the page (`https://medium.ngtv.io/v2/media/...`) and it doesn't have any! In fact, throwing it into yt-dlp without any additional parameters seems to work without any issues at all.

```
[debug] Command-line config: ['--verbose', 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/master_cl_de.m3u8']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.07.18 [135f05e]
[debug] Python 3.10.6 (CPython 64bit) - Linux-5.19.2-arch1-1-x86_64-with-glibc2.36 (glibc 2.36)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2022.06.15, mutagen-1.45.1, secretstorage-3.3.2, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/master_cl_de.m3u8
[generic] master_cl_de: Downloading webpage
[debug] Identified a direct video link
[generic] master_cl_de: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] master_cl_de: Downloading 1 format(s): 8138+audio-ENGLISH
[debug] Invoking hlsnative downloader on "https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/5cfdca83-7b48-49e5-b9fe-0902ec725b88-565823-1920x1080_8000k.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 305
[download] Destination: master_cl_de [master_cl_de].f8138.mp4
[download]   2.3% of ~1.50GiB at    2.69MiB/s ETA 12:05 (frag 7/305)
```

**Comment by 89z** [comment]
This isn't YT-DLP repo.

**Comment by tacohitbox** [comment]
OH my bad, i seriously thought i was there. 
It still works in youtube-dl though!
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/master_cl_de.m3u8']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.10.6 (CPython) - Linux-5.19.2-arch1-1-x86_64-with-glibc2.36
[debug] exe versions: ffmpeg 5.1, ffprobe 5.1, rtmpdump 2.4
[debug] Proxy map: {}
[generic] master_cl_de: Requesting header
[generic] master_cl_de: Downloading m3u8 information
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/5cfdca83-7b48-49e5-b9fe-0902ec725b88-565823-1920x1080_8000k.m3u8'
[download] Destination: master_cl_de-master_cl_de.f8138.mp4
[debug] ffmpeg command line: ffmpeg -y -loglevel verbose -headers 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.39 Safari/537.36
Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-us,en;q=0.5
' -i https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/5cfdca83-7b48-49e5-b9fe-0902ec725b88-565823-1920x1080_8000k.m3u8 -c copy -f mp4 file:master_cl_de-master_cl_de.f8138.mp4.part
ffmpeg version n5.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 12.1.1 (GCC) 20220730
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-amf --enable-avisynth --enable-cuda-llvm --enable-lto --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libdav1d --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-librav1e --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-libzimg --enable-nvdec --enable-nvenc --enable-shared --enable-version3
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
[tcp @ 0x563773d91f40] Starting connection attempt to 23.196.47.80 port 443
[tcp @ 0x563773d91f40] Successfully connected to 23.196.47.80 port 443
[hls @ 0x563773d8dcc0] Skip ('#EXT-X-VERSION:6')
[hls @ 0x563773d8dcc0] HLS request for url 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/h264_1920x1080_8000k.mp4', offset 0, playlist 0
[hls @ 0x563773d8dcc0] Opening 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/h264_1920x1080_8000k.mp4' for reading
[tcp @ 0x5637743c8680] Starting connection attempt to 23.196.47.107 port 443
[tcp @ 0x5637743c8680] Successfully connected to 23.196.47.107 port 443
[AVIOContext @ 0x5637743c4140] Statistics: 876 bytes read, 0 seeks
[hls @ 0x563773d8dcc0] HLS request for url 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_1.m4s', offset 0, playlist 0
[hls @ 0x563773d8dcc0] Opening 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_1.m4s' for reading
[tcp @ 0x5637743bfc40] Starting connection attempt to 23.196.47.80 port 443
[tcp @ 0x5637743bfc40] Successfully connected to 23.196.47.80 port 443
[hls @ 0x563773d8dcc0] HLS request for url 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_2.m4s', offset 0, playlist 0
[hls @ 0x563773d8dcc0] Opening 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_2.m4s' for reading
[tcp @ 0x563773e331c0] Starting connection attempt to 23.196.47.107 port 443
[tcp @ 0x563773e331c0] Starting connection attempt to 2600:1405:1000::17c4:2f39 port 443
[tcp @ 0x563773e331c0] Starting connection attempt to 23.196.47.80 port 443
[tcp @ 0x563773e331c0] Successfully connected to 23.196.47.107 port 443
[h264 @ 0x56377439ea40] Reinit context to 1920x1088, pix_fmt: yuv420p
Input #0, hls, from 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/5cfdca83-7b48-49e5-b9fe-0902ec725b88-565823-1920x1080_8000k.m3u8':
  Duration: 00:30:27.83, start: 0.000000, bitrate: 0 kb/s
  Program 0 
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (High), 1 reference frame (avc1 / 0x31637661), yuv420p(left), 1920x1080 (1920x1088) [SAR 1:1 DAR 16:9], Closed Captions, 8216 kb/s, 29.97 fps, 29.97 tbr, 30k tbn (default)
    Metadata:
      variant_bitrate : 0
      major_brand     : isom
      minor_version   : 0
      compatible_brands: iso8mp41dashavc1cmfc
      creation_time   : 2021-12-13T21:30:51.000000Z
Output #0, mp4, to 'file:master_cl_de-master_cl_de.f8138.mp4.part':
  Metadata:
    encoder         : Lavf59.27.100
  Stream #0:0: Video: h264 (High), 1 reference frame (avc1 / 0x31637661), yuv420p(left), 1920x1080 (0x0) [SAR 1:1 DAR 16:9], q=2-31, 8216 kb/s, 29.97 fps, 29.97 tbr, 30k tbn (default)
    Metadata:
      variant_bitrate : 0
      major_brand     : isom
      minor_version   : 0
      compatible_brands: iso8mp41dashavc1cmfc
      creation_time   : 2021-12-13T21:30:51.000000Z
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
Press [q] to stop, [?] for help
[hls @ 0x563773d8dcc0] HLS request for url 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_3.m4s', offset 0, playlist 0
[https @ 0x5637743c5780] Opening 'https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/segment_3.m4s' for reading
^Cframe=  359 fps= 42 q=-1.0 Lsize=   11883kB time=00:00:11.91 bitrate=8172.1kbits/s speed=1.41x    
video:11879kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.036609%
Input file #0 (https://clips-media-aka.warnermediacdn.com/hbomax/clips/2021-12/565823-3464db935b394afbba5ce3ff9e375f6c/cmaf/h264_1920x1080_8000k/5cfdca83-7b48-49e5-b9fe-0902ec725b88-565823-1920x1080_8000k.m3u8):
  Input stream #0:0 (video): 359 packets read (12163707 bytes); 
  Total: 359 packets (12163707 bytes) demuxed
Output file #0 (file:master_cl_de-master_cl_de.f8138.mp4.part):
  Output stream #0:0 (video): 359 packets muxed (12163707 bytes); 
  Total: 359 packets (12163707 bytes) muxed
[AVIOContext @ 0x563773dfbcc0] Statistics: 12168164 bytes written, 2 seeks, 50 writeouts
[AVIOContext @ 0x5637743c5340] Statistics: 6006705 bytes read, 0 seeks
[AVIOContext @ 0x5637743cabc0] Statistics: 6170724 bytes read, 0 seeks
[AVIOContext @ 0x5637743b80c0] Statistics: 9478 bytes read, 0 seeks
```
